user.go

```diff
@@ -29,17 +29,19 @@ type users struct {
 
 // User represents a Terraform Enterprise user.
 type User struct {
-	ID               string           `jsonapi:"primary,users"`
-	AvatarURL        string           `jsonapi:"attr,avatar-url"`
-	Email            string           `jsonapi:"attr,email"`
-	IsServiceAccount bool             `jsonapi:"attr,is-service-account"`
-	TwoFactor        *TwoFactor       `jsonapi:"attr,two-factor"`
-	UnconfirmedEmail string           `jsonapi:"attr,unconfirmed-email"`
-	Username         string           `jsonapi:"attr,username"`
-	V2Only           bool             `jsonapi:"attr,v2-only"`
-	IsSiteAdmin      *bool            `jsonapi:"attr,is-site-admin"`
-	IsSsoLogin       *bool            `jsonapi:"attr,is-sso-login"`
-	Permissions      *UserPermissions `jsonapi:"attr,permissions"`
+	ID               string     `jsonapi:"primary,users"`
+	AvatarURL        string     `jsonapi:"attr,avatar-url"`
+	Email            string     `jsonapi:"attr,email"`
+	IsServiceAccount bool       `jsonapi:"attr,is-service-account"`
+	TwoFactor        *TwoFactor `jsonapi:"attr,two-factor"`
+	UnconfirmedEmail string     `jsonapi:"attr,unconfirmed-email"`
+	Username         string     `jsonapi:"attr,username"`
+	V2Only           bool       `jsonapi:"attr,v2-only"`
+	// Deprecated: IsSiteAdmin was deprecated in v202406 and will be removed in a future version of Terraform Enterprise
```

Comment:
The version will actually be known as v202406-1... and also I think it would be best to mention the new field. For clarity, here's how this godoc comment shows up.

![Screenshot 2024-05-15 at 2 33 34 PM](https://github.com/hashicorp/go-tfe/assets/174332/1d3dabae-de8a-4d7e-abae-5e2a157c6b15)

```suggestion
	// Deprecated: IsSiteAdmin was deprecated in Terraform Enterprise v202406-1
	// and will be removed in a future version. Use IsAdmin Instead.
```